ste/mgr-JobPartTransferMgr.go

```diff
@@ -279,6 +279,16 @@ func (jptm *jobPartTransferMgr) Info() TransferInfo {
 	// does not exceeds 50000 (max number of block per blob)
 	if blockSize == 0 {
 		blockSize = uint32(common.DefaultBlockBlobBlockSize)
+		//Workaround to support transfer of small files from GCP.
+		//This workaround is in place because GCP sends back a Response code 200 instead of Response code 209
+		//when file size is smaller than the blockSize.
+		if int64(blockSize) > sourceSize && sourceSize != 0 && plan.FromTo == common.EFromTo.GCPBlob() {
+			if sourceSize < 0 {
+				panic(fmt.Errorf("GCP sourceSize cannot be negative\n"))
+			} else {
+				blockSize = uint32(sourceSize) / 2
```

Comment:
>uint32(sourceSize) / 2 [](start = 16, length = 22)

doing 2 calls has a perf impact, can we do 1 call with the range taken off?

Reply:
Resolving, outdated